### PR TITLE
This adds many missing API endpoints - fixes #641

### LINKED ISSUE #641: update the lookup module for new/missing api endpoints.
##### ISSUE TYPE

 - Feature Idea

###### NetBox:

Any netbox > 2.11.0

###### Collection:

looked at v3.3.0 

##### SUMMARY

The lookup module is missing several extra endpoints:

```
{
    "config-contexts": "http://netbox.nersc.gov/api/extras/config-contexts/",
    "content-types": "http://netbox.nersc.gov/api/extras/content-types/",
    "custom-fields": "http://netbox.nersc.gov/api/extras/custom-fields/",
    "custom-links": "http://netbox.nersc.gov/api/extras/custom-links/",
    "export-templates": "http://netbox.nersc.gov/api/extras/export-templates/",
    "image-attachments": "http://netbox.nersc.gov/api/extras/image-attachments/",
    "job-results": "http://netbox.nersc.gov/api/extras/job-results/",
    "journal-entries": "http://netbox.nersc.gov/api/extras/journal-entries/",
    "object-changes": "http://netbox.nersc.gov/api/extras/object-changes/",
    "reports": "http://netbox.nersc.gov/api/extras/reports/",
    "scripts": "http://netbox.nersc.gov/api/extras/scripts/",
    "tags": "http://netbox.nersc.gov/api/extras/tags/",
    "webhooks": "http://netbox.nersc.gov/api/extras/webhooks/"
}
```

vs.

```
lookup]$ grep extra nb_lookup.py 
        "config-contexts": {"endpoint": netbox.extras.config_contexts},
        "graphs": {"endpoint": netbox.extras.graphs},
        "image-attachments": {"endpoint": netbox.extras.image_attachments},
        "object-changes": {"endpoint": netbox.extras.object_changes},
        "reports": {"endpoint": netbox.extras.reports},
        "tags": {"endpoint": netbox.extras.tags},
        "topology-maps": {"endpoint": netbox.extras.topology_maps},
```

so that's missing:
- content-types
- custom-fields
- custom-links
- export-templates
- job-results
- journal-entries
- scripts
- webhooks

Of the above missing endpoints, I'm not sure which ones are still supported.

### PR DESCRIPTION
**Pull request by ThomasADavis**
Fixes issue #641

It include endpoints for v3.1

No depecrated API endpoints have been removed.

### CODE REVIEW

File: plugins/lookup/nb_lookup.py [hunk 1]
```diff
@@ -215,20 +235,29 @@ def get_endpoint(netbox, term):
...
         "topology-maps": {"endpoint": netbox.extras.topology_maps},
+        "users": {"endpoint": netbox.users.users},
         "virtual-chassis": {"endpoint": netbox.dcim.virtual_chassis},
         "virtual-machines": {"endpoint": netbox.virtualization.virtual_machines},
         "virtualization-interfaces": {"endpoint": netbox.virtualization.interfaces},
         "vlan-groups": {"endpoint": netbox.ipam.vlan_groups},
         "vlans": {"endpoint": netbox.ipam.vlans},
         "vrfs": {"endpoint": netbox.ipam.vrfs},
+        "webhooks": {"endpoint": netbox.extras.webhooks},
+        # "wireless-lan-groups": {"endpoint": netbox.wireless.wireless_lan_groups},
```
Comment (sc68cal) [inline plugins/lookup/nb_lookup.py:258]: Do not add this to a changeset if you are going to leave it commented out, IMO.

**Review by sc68cal — CHANGES_REQUESTED** [review]

**Comment by rodvand** [comment]
How about you just add missing endpoints, and leave out the 3.1 endpoints for now? And then when 3.1 is released we'll cut a specific release where we add 3.1 stuff? As you mentioned in another issue we also have to wait for a pynetbox to support some of the endpoints.

**Comment by ThomasADavis** [comment]
I'm thinking of this:

```
    try:
        netbox_endpoint_map["wireless-lan-groups"] = {"endpoint": netbox.wireless.wireless_lan_groups}
        netbox_endpoint_map["wireless-lan-groups"] = {"endpoint": netbox.wireless.wireless_lan_groups}
        netbox_endpoint_map["wireless-lan"] = {"endpoint": netbox.wireless.wireless_lan}
        netbox_endpoint_map["wireless-links"] = {"endpoint": netbox.wireless.wireless_links}

    except AttributeError as e:
        Display().v(u"no wireless API support, please update your pynetbox.")

```
so it will still work on older versions of pynetbox.

**Review by sc68cal — APPROVED** [review]
looks good to me

**Comment by ThomasADavis** [comment]
The wireless api can't be added until widespread acceptance of an updated pynetbox is available..which means when they get added, it could break many ansible installations.  For me, widespread acceptance means when a distribution like Centos8 streamCentos8/RHEL8/RHEL9,  Fedora ships the updated pynetbox as an RPM. We tried a python venv environment for a year, and the issues was numerous (starting with python it self)

Using the try/exception avoids that problem.

**Comment by rodvand** [comment]
I can see how we would need some logic to deal with new endpoints with NetBox 3.1. I guess it has just been so long since the endpoints have changed it hasn't really been an issue.

**Pull request merged**